src/GeneralStateTestsFiller/stReturnDataTest/revertRetDataSizeFiller.yml

```diff
@@ -0,0 +1,365 @@
+    currentGasLimit: 100000000
+    currentNumber: 1
+    currentTimestamp: 1000
+    previousHash: 5e20a0453cecd065ea59c37ac63e079ee08998b6045136a8ce6635c7912ec0b6
+
+  _info:
+    comment: Ori Pomerantz qbzzt1@gmail.com
+
+  pre:
+
+    cccccccccccccccccccccccccccccccccccccccc:
+      balance: '0x0ba1a9ce0ba1a9ce'
+      code: |
+        {   ;  $4 is the type of thing that fails
+            ; $36 is the failure itself
+
+            (def 'callType   $4)
+            (def 'call         0xf1)
+            (def 'callcode     0xf2)
+            (def 'delegatecall 0xf4)
+            (def 'staticcall   0xfa)
+            (def 'create       0xf0)
+            (def 'create2      0xf5)
+
+            (def 'failureType $36)
+            (def 'oog 0)
+
+            ; We need these values for CREATE(2)
+            (def 'uf        0x200)
+            (def 'jmp       0x300)
+            (def 'jmpi      0x400)
+            (def 'badOpcode 0x500)
+
+
+            (def 'NOP 0)
+
+            ; Code for CREATE(2) to fail
+           
+            (def 'codeLoc      0x0000)
+            (def 'codeLength   0x0100)
+        
+            (if (= failureType oog) 
+               [codeLength] (lll (sha3 0 (- 0 1)) codeLoc)
+               NOP
+            )
+
+            ; lll protects us from the other failures, so we need to use vm code
+            (if (= failureType uf) {
+                 (mstore8 codeLoc         0x50)     ; POP
+                 (mstore8 (+ codeLoc 1)   0x00)     ; STOP
+                 [codeLength] 2
+               }
+               NOP
+            )
+
+            (if (= failureType jmp) {
+                 (mstore8 codeLoc         0x60)     ; PUSH1 0x00                
+                 (mstore8 (+ codeLoc 1)   0x00) 
+                 (mstore8 (+ codeLoc 2)   0x56)     ; JUMP
+                 (mstore8 (+ codeLoc 2)   0x00)     ; STOP
+                 [codeLength] 4
+               }
+               NOP
+            )
+
+            (if (= failureType jmpi) {
+                 (mstore8 codeLoc         0x60)     ; PUSH1 0x01                
+                 (mstore8 (+ codeLoc 1)   0x01) 
+                 (mstore8 (+ codeLoc 2)   0x60)     ; PUSH1 0x01
+                 (mstore8 (+ codeLoc 3)   0x01) 
+                 (mstore8 (+ codeLoc 4)   0x57)     ; JUMPI
+                 (mstore8 (+ codeLoc 5)   0x00)     ; STOP
+                 [codeLength] 6
+               }
+               NOP
+            )
+            
+
+
+
+            (if (= failureType badOpcode) {
+                 (mstore8 codeLoc         0xFE)     ; Invalid opcode
+                 (mstore8 (+ codeLoc 1)   0x00)     ; STOP 
+                 [codeLength] 2
+               }
+               NOP
+            )
+            
+
+
+
+            ; The actual failures start here ----------------------------------------
+
+            (if (and (= callType call) (= failureType oog))
+                  {
+                    (call (gas) 0x100 0 0 0 0 0x40)
```

Comment:
where the oog happens in this case? 
when `[codeLength] (lll (sha3 0 (- 0 1)) codeLoc)`  is executed?

Reply:
```
[codeLength] (lll (sha3 0 (- 0 1)) codeLoc)
```

Doesn't execute the `sha3`, it writes the compiled LLL code to `codeLoc` and then writes the length in `codeLength`. 

In the case of `-d 0` (CALL and OOG), the OOG happens inside the second 0x100 contract call, the one on line 106. I am doing two calls so the first will create a RETURNDATASIZE for the second to overwrite.

```
                    (call (gas) 0x100 0 0 0 0 0x40)
                    [[0]] (returndatasize)
                    (call 0 0x100 0 0 0 0 0x40)
                    [[1]] (returndatasize)
```

To verify this you can look at the attached `--vmtrace`. The successful return is on line 103:

```
     92-2          PUSH1         3   1252575  78726093
     93-2          PUSH1         3   1252578  78726090
     94-2         RETURN         0   1252581  78726087
    ^[[33m          RETURN 0x7ffffffffffffffffffffffffffffffffffffffffffffffffffff$
95-1            POP         2      2960  79975708
96-1 RETURNDATASIZE         2      2962  79975706
97-1          PUSH1         3      2964  79975704
98-1         SSTORE      5000      2967  79975701
    ^[[33m          SSTORE [0x0] = 0x40^[[0m
```

The failure is line 128:
```
104-1          PUSH2         3      7982  79970686
105-1          PUSH1         3      7985  79970683
106-1           CALL       100      7988  79970680
             ^[[33mSUBCALL: 0x100^[[0m
    107-2          PUSH1         3  79978668         0          out of gas
108-1            POP         2      8088  79970580
109-1 RETURNDATASIZE         2      8090  79970578
110-1          PUSH1         3      8092  79970576
111-1         SSTORE      5000      8095  79970573
    ^[[33m          SSTORE [0x1] = 0x0^[[0m
```


[revertRetDataSize.txt](https://github.com/ethereum/tests/files/6174685/revertRetDataSize.txt)